renderlib/ClipPlaneTool.cpp

```diff
@@ -0,0 +1,67 @@
+#include "ClipPlaneTool.h"
+
+#include "AppScene.h"
+#include "Logging.h"
+#include "SceneView.h"
+
+void
+ClipPlaneTool::action(SceneView& scene, Gesture& gesture)
+{
+}
+
+void
+ClipPlaneTool::draw(SceneView& scene, Gesture& gesture)
+{
+  if (!scene.scene) {
+    return;
+  }
+
+  // TODO draw this as an oriented grid centered in the view (or on the volume?)
+  glm::vec3 n = m_plane.normal;
+  // where to draw the plane
+  glm::vec3 p = m_pos; // m_plane->normal * -m_plane->d;
+  glm::vec3 u = glm::cross(n, glm::vec3(0, 1, 0));
+  if (glm::length(u) < 0.001f) {
+    u = glm::normalize(glm::cross(n, glm::vec3(1, 0, 0)));
+  } else {
+    u = glm::normalize(u);
+  }
+  glm::vec3 v = glm::normalize(glm::cross(n, u));
+  u = glm::normalize(glm::cross(v, n));
```

Comment:
Why do you recalculate `u` here?

Reply:
You generally have to recalculate u because the first u is not guaranteed perpendicular to BOTH of v and n.